homeassistant/components/device_tracker/freebox.py

```diff
@@ -0,0 +1,141 @@
+
+
+async def async_setup_scanner(hass, config, async_see, discovery_info=None):
+    """Set up the Freebox device tracker and start the polling."""
+    freebox_config = copy.deepcopy(config)
+    if discovery_info is not None:
+        freebox_config[CONF_HOST] = discovery_info['properties']['api_domain']
+        freebox_config[CONF_PORT] = discovery_info['properties']['https_port']
+        _LOGGER.info("Discovered Freebox server: %s:%s",
+                     freebox_config[CONF_HOST], freebox_config[CONF_PORT])
+
+    scanner = FreeboxDeviceScanner(hass, freebox_config, async_see)
+    interval = freebox_config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+    await scanner.async_start(hass, interval)
+    return True
+
+
+Device = namedtuple('Device', ['id', 'name', 'ip'])
+
+
+def _build_device(device_dict):
+    return Device(
+        device_dict['l2ident']['id'],
+        device_dict['primary_name'],
+        device_dict['l3connectivities'][0]['addr'])
+
+
+class FreeboxDeviceScanner(object):
+    """This class scans for devices connected to the Freebox."""
+
+    def __init__(self, hass, config, async_see):
+        """Initialize the scanner."""
+        self.host = config[CONF_HOST]
+        self.port = config[CONF_PORT]
+        self.token_file = hass.config.path(FREEBOX_CONFIG_FILE)
+
+        self.async_see = async_see
+
+    async def async_start(self, hass, interval):
+        """Perform a first update and start pooling at the given interval."""
+        await self.async_update_info()
+        async_track_time_interval(hass, self.async_update_info, interval)
+
+    @Throttle(MIN_TIME_BETWEEN_SCANS)
+    async def async_update_info(self, now=None):
+        """Check the Freebox for devices."""
+        _LOGGER.info('Scanning devices')
+
+        from aiofreepybox import Freepybox
+        import asyncio
+        import socket
+
+        # Hardcode the app description to avoid invalidating the authentication
+        # file at each new version.
+        # The version can be changed if we want the user to re-authorize HASS
+        # on her Freebox.
+        app_desc = {
+            'app_id': 'hass',
+            'app_name': 'Home Assistant',
+            'app_version': '0.65',
+            'device_name': socket.gethostname()
+        }
+
+        api_version = 'v1'  # Use the lowest working version.
+        fbx = Freepybox(
+            app_desc=app_desc,
+            token_file=self.token_file,
+            api_version=api_version)
+        await fbx.open(self.host, self.port)
+        try:
+            hosts = await fbx.lan.get_hosts_list()
+        finally:
+            # Initiate closing, will be awaited later
+            closing_task = asyncio.ensure_future(fbx.close())
+
+        last_results = [_build_device(device)
+                        for device in hosts
+                        if device['active']]
+
+        asyncio.gather(*[self.async_see(mac=d.id, host_name=d.name)
```

Comment:
The future that this returns is never awaited.

Reply:
Since we're not interested in the result of the future, ie the return value of `async_see`, I suggest you use `asyncio.wait` instead of `asyncio.gather`.

Reply:
Right, fixed